lib/src/validator/changelog.dart

```diff
@@ -19,10 +19,16 @@ class ChangelogValidator extends Validator {
       final changelog = entrypoint.root.changelogPath;
 
       if (changelog == null) {
-        // No changelog was found, which is fine. Return with no warnings.
+        warnings.add('Please add a`CHANGELOG.md` to your package. '
+            'See https://dart.dev/tools/pub/publishing#important-files.');
         return;
       }
 
+      if (!changelog.endsWith('CHANGELOG.md')) {
```

Comment:
@sigurdm, why do we tests using `endsWith`? Can't we use extract the filename from the path and test for equality?

Reply:
Sure - that would be better. I'll make a fix